pypesto/objective/petab_import.py

```diff
@@ -341,58 +357,61 @@ def rdatas_to_measurement_df(self, rdatas, model=None):
         return df
 
 
-def _get_simulation_conditions(condition_df, measurement_df):
-    """
-    Compute the conditions by which to group the measurements, so
-    that for each group an amici.ExpData can be generated.
-
-    Returns
-    -------
-    (grouping_cols, simulation_conditions): tuple
-        Here, grouping_cols is the columns according to which the grouping
-        was done, and simulation_conditions are the identified conditions.
+def _check_parameter_mapping_ok(
+        mapping_par_opt_to_par_sim, par_sim_ids, model, edatas):
     """
-    # make sure index is reset
-    condition_df = condition_df.reset_index()
+    Check whether there are suspicious parameter mappings and/or data points.
 
-    # find columns to group by (i.e. if not all nans).
-    # number of amici simulations will be number of unique
-    # (preequilibrationCondition, simulationCondition) pairs.
-    # can be improved by checking for identical condition vectors.
-    grouping_cols = petab.core.get_notnull_columns(
-        measurement_df,
-        ['simulationConditionId', 'preequilibrationConditionId'])
-
-    # group by cols and return dataframe containing each combination
-    # of those rows only once (and an additional counting row)
-    simulation_conditions = measurement_df.groupby(
-        grouping_cols).size().reset_index()
-
-    return grouping_cols, simulation_conditions
-
-
-def _get_rows_for_condition(measurement_df, grouping_cols, condition):
+    Currently checks whether nan values in the parameter mapping table
+    corresponds to nan columns in the edatas, corresponding to missing
+    data points.
     """
-    Extract rows in `measurement_df` according to `grouping_cols`
-    for `condition`.
-
-    Returns
-    -------
+    # regular expression for noise and observable parameters
+    rex = "(noise|observable)Parameter[0-9]+_"
```

Comment:
Feel free to re.compile

Reply:
yes, it's better re.compile than to recompile every time. thanks!